### PR TITLE
Limit result stack

### PR DESCRIPTION
**Pull request by shargon**
Alternative to https://github.com/neo-project/neo-modules/pull/695
Close https://github.com/neo-project/neo/issues/2666

### CODE REVIEW

**Comment by Liaojinghui** [comment]
Please also take care

https://github.com/neo-project/neo-express/blob/bbde1ff6b932277fb7ffd5313b12e784edb16d56/src/neoxp/Node/OfflineNode.cs#L127

https://github.com/neo-project/neo-node/blob/115e189008f112a001c651e6737e2241f35c5e0b/neo-gui/GUI/InvokeContractDialog.cs#L93

https://github.com/neo-project/neo-modules/blob/43298b065b2c42526f1c7f7bd691f4a79bc077c0/src/RpcServer/RpcServer.SmartContract.cs#L78

https://github.com/neo-project/neo-gui-2.x/blob/b816efbadc9bc731bf5ccd4cb545303bd39aabfa/neo-gui/UI/InvokeContractDialog.cs#L188

https://github.com/neo-project/neo-node/blob/24bd2ccfb7abd6ccdcb2e539a9896d1c3b02a4a7/neo-cli/CLI/MainService.cs#L655

https://github.com/neo-project/neo-modules/blob/e99d099fd3ae87e7cf6921f7f8e8f1d94f710dab/src/RpcServer/RpcServer.Wallet.cs#L382

https://github.com/neo-project/neo-debugger/blob/bc68ce1b61f5cbcd1ba60f51f1bccc133613f85e/src/adapter2/DebugSession.cs#L480

**Comment by Liaojinghui** [comment]
May you please move function `ParseStack` to `neo core/VM/helper.cs` such that it can be used every where.

**Review by superboyiii — DISMISSED** [review]
Test: **PASS**
Successfully synced to the latest block and haven't been stuck.
invoke applicationlog for tx:`0x40e171249642639be3e54dd6acfe5948b1d3f240f505e7a3403cdc9bd3e1068a`  and get the expected response:
```
{"jsonrpc":"2.0","id":1,"result":{"txid":"0x40e171249642639be3e54dd6acfe5948b1d3f240f505e7a3403cdc9bd3e1068a","executions":[{"trigger":"Application","vmstate":"HALT","exception":null,"gasconsumed":"13053","stack":"error: max stack size reached","notifications":[]}]}}
```

**Comment by erikzhang** [comment]
Please check neo-project/neo#2671

**Comment by Liaojinghui** [comment]
Hey shargon, the transaciton to issue a 100k size newbuffer costs 0.0000771GAS, this means an attacker can use 0.78GAS to generate 1Gb Log in 1024 attack transactions.

**Comment by Liaojinghui** [comment]
Hey shargon, the transaciton to issue a 100k size newbuffer costs 0.0000771GAS, this means an attacker can use 0.78GAS to generate 1Gb Log in 1024 attack transactions.

**Review by superboyiii — APPROVED** [review]

**Comment by superboyiii** [comment]
@erikzhang Merge?

**Comment by Liaojinghui** [comment]
https://github.com/neo-project/neo-express/blob/bbde1ff6b932277fb7ffd5313b12e784edb16d56/src/neoxp/Node/OfflineNode.cs#L127

https://github.com/neo-project/neo-node/blob/115e189008f112a001c651e6737e2241f35c5e0b/neo-gui/GUI/InvokeContractDialog.cs#L93

https://github.com/neo-project/neo-modules/blob/43298b065b2c42526f1c7f7bd691f4a79bc077c0/src/RpcServer/RpcServer.SmartContract.cs#L78
https://github.com/neo-project/neo-gui-2.x/blob/b816efbadc9bc731bf5ccd4cb545303bd39aabfa/neo-gui/UI/InvokeContractDialog.cs#L188

https://github.com/neo-project/neo-node/blob/24bd2ccfb7abd6ccdcb2e539a9896d1c3b02a4a7/neo-cli/CLI/MainService.cs#L655

https://github.com/neo-project/neo-modules/blob/e99d099fd3ae87e7cf6921f7f8e8f1d94f710dab/src/RpcServer/RpcServer.Wallet.cs#L382

https://github.com/neo-project/neo-debugger/blob/bc68ce1b61f5cbcd1ba60f51f1bccc133613f85e/src/adapter2/DebugSession.cs#L480

**Comment by erikzhang** [comment]
Update `config.json` too?

File: src/Directory.Build.props [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Neo" Version="3.1.0-CI01329" />
```
Comment (erikzhang) [inline src/Directory.Build.props:17]: We need to wait for the neo master branch.

**Comment by superboyiii** [comment]
> These also need update @shargon
> 
> https://github.com/neo-project/neo-express/blob/bbde1ff6b932277fb7ffd5313b12e784edb16d56/src/neoxp/Node/OfflineNode.cs#L127
> 
> https://github.com/neo-project/neo-node/blob/115e189008f112a001c651e6737e2241f35c5e0b/neo-gui/GUI/InvokeContractDialog.cs#L93
> 
> https://github.com/neo-project/neo-modules/blob/43298b065b2c42526f1c7f7bd691f4a79bc077c0/src/RpcServer/RpcServer.SmartContract.cs#L78
> 
> 
> https://github.com/neo-project/neo-gui-2.x/blob/b816efbadc9bc731bf5ccd4cb545303bd39aabfa/neo-gui/UI/InvokeContractDialog.cs#L188
> https://github.com/neo-project/neo-node/blob/24bd2ccfb7abd6ccdcb2e539a9896d1c3b02a4a7/neo-cli/CLI/MainService.cs#L655
> 
> https://github.com/neo-project/neo-modules/blob/e99d099fd3ae87e7cf6921f7f8e8f1d94f710dab/src/RpcServer/RpcServer.Wallet.cs#L382
> 
> https://github.com/neo-project/neo-debugger/blob/bc68ce1b61f5cbcd1ba60f51f1bccc133613f85e/src/adapter2/DebugSession.cs#L480

@shargon Could you update these?

**Review by erikzhang — APPROVED** [review]

**Comment by Liaojinghui** [comment]
Are we going to merge this without fixing [neo-modules/src/RpcServer/RpcServer.Wallet.cs](https://github.com/neo-project/neo-modules/blob/e99d099fd3ae87e7cf6921f7f8e8f1d94f710dab/src/RpcServer/RpcServer.Wallet.cs#L382) and [neo-modules/src/RpcServer/RpcServer.Wallet.cs](https://github.com/neo-project/neo-modules/blob/e99d099fd3ae87e7cf6921f7f8e8f1d94f710dab/src/RpcServer/RpcServer.Wallet.cs#L382)?

**Comment by superboyiii** [comment]
@erikzhang Could you have a review?

**Review by Liaojinghui — DISMISSED** [review]

**Review by erikzhang — APPROVED** [review]

**Review by Liaojinghui — APPROVED** [review]

**Pull request merged**